adafruit_wiznet5k/adafruit_wiznet5k.py

```diff
@@ -175,6 +175,17 @@ def __init__(
         self.mac_address = mac
         self._src_port = 0
         self._dns = 0
+
+        # First wait link status is on
+        # to avoid the code during DHCP - assert self.link_status, "Ethernet cable disconnected!"
+        start_time = time.monotonic()
+        while True:
+            if self.link_status or ((time.monotonic() - start_time) > 5) :
```

Comment:
I do not understand why this is required here. We're checking link status within most `socket_` functions and already checking if the hardware is initialized further up in `_w5100_init`. Could you clarify why you want this added?

Reply:
I know, socket_ functions already check the link. so, I always got the message without this block if I enable DHCP.
"AssertionError: Ethernet cable disconnected!"
Because, before PHY link setup, code run the WIZNET5K() -> DHCP start -> DHCP sending message -> AssertionError.
```
# Initialize ethernet interface with DHCP
eth = WIZNET5K(spi_bus, cs, is_dhcp=True, mac=MY_MAC, debug=False)
print("Chip Version:", eth.chip)
print("MAC Address:", [hex(i) for i in eth.mac_address])
print("My IP address is:", eth.pretty_ip(eth.ip_address))
```
Without this (or some wait time before calling WIZNET5K() ) we got the AssertionError message only (specially, I controlled RESET pin before calling WIZNET5K()). so I added this code.
if you don't, I think all main code need to check return value of WIZNET5K() instead of this code block.

PS) In case of manual IP setting, there is no problem without this block. Because there is no sending, receiving events for setting IP address.

Reply:
Could we check this within `set_dhcp` instead?

Reply:
Good idea. I move this into ``set_dhcp``